extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/registry/json/JsonMeterRegistry.java

```diff
@@ -0,0 +1,118 @@
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.cumulative.CumulativeCounter;
+import io.micrometer.core.instrument.cumulative.CumulativeDistributionSummary;
+import io.micrometer.core.instrument.cumulative.CumulativeFunctionCounter;
+import io.micrometer.core.instrument.cumulative.CumulativeFunctionTimer;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.internal.DefaultGauge;
+import io.micrometer.core.instrument.internal.DefaultMeter;
+import io.micrometer.core.instrument.noop.NoopLongTaskTimer;
+
+/**
+ * A registry that, when exported, mimics the JSON exporter from MP Metrics 3.0
+ * as closely as it is reasonable to attempt to.
+ */
+public class JsonMeterRegistry extends MeterRegistry {
+
+    private final JsonExporter jsonExporter;
+
+    public JsonMeterRegistry(Clock clock) {
+        super(clock);
+        this.jsonExporter = new JsonExporter();
+    }
+
+    @Override
+    protected <T> Gauge newGauge(Meter.Id id, T obj, ToDoubleFunction<T> valueFunction) {
+        return new DefaultGauge<>(id, obj, valueFunction);
+    }
+
+    @Override
+    protected Counter newCounter(Meter.Id id) {
+        return new CumulativeCounter(id);
+    }
+
+    @Override
+    protected Timer newTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig,
+            PauseDetector pauseDetector) {
+        // turn off percentiles and histograms because we don't need them for the JSON export
+        distributionStatisticConfig = DistributionStatisticConfig.builder()
+                .percentilesHistogram(false)
+                .percentiles(new double[0])
+                .build()
+                .merge(distributionStatisticConfig);
+        return new JsonTimer(id, clock, distributionStatisticConfig, pauseDetector, getBaseTimeUnit());
+    }
+
+    @Override
+    protected DistributionSummary newDistributionSummary(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig,
+            double scale) {
+        distributionStatisticConfig = DistributionStatisticConfig.builder()
+                .percentilePrecision(3)
+                .percentilesHistogram(false)
+                .minimumExpectedValue(Double.MIN_VALUE)
+                .maximumExpectedValue(Double.MAX_VALUE)
+                .bufferLength(1024)
+                .expiry(Duration.ofDays(3))
+                .build()
```

Comment:
I would like to understand where these values come from. Negative values don't seem correct for an out-of-the-box default. Can we talk through this?

Reply:
The expiry and bufferLength, I just guessed. Perhaps they should be configurable?

Not sure what you mean by negative values. There are no negative values. `Double.MIN_VALUE` is greater than zero.

Reply:
You're right. I forgot that Double doesn't work that way.

Reply:
Mark this one resolved. ;)